src/Service/ProductBundleCreatorInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace solutionDrive\SyliusProductBundlesPlugin\Service;
+
+use solutionDrive\SyliusProductBundlesPlugin\Entity\ProductBundleInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+
+/**
+ * Created by solutionDrive GmbH
+ *
+ * @copyright 2018 solutionDrive GmbH
+ */
+interface ProductBundleCreatorInterface
+{
+    public function createProductBundle(
+        string $productBundleName,
+        ProductInterface $productBundleProduct
+    ): ProductBundleCreator;
+
+    public function getProductBundle(): ProductBundleInterface;
+
+    public function addSlot(string $slotName, array $options = [], array $products = []): ProductBundleCreator;
```

Comment:
Coding Style: If you declare arrays, an annotation with typhints of the form ```@param ProductInterface[] $products``` would be useful.

Reply:
Indeed. For $products this should be done. 
For $options I'm at the moment thinking about pros & cons of introducing an Options-Object, because it's quite hard to communicate the possible Options besides of writing them in the README.

Reply:
Good point. Example: Each time I'm writing Symfony Forms I'm irritated by the $options array, because it's very uncomfortable to find the correct possible options.
I'd really appreciate a meaningful configuration class, if possible.

(In case of my example of Symfony Forms I think it is by design not possible because of the many different types with many different interfaces... )

Reply:
I had exactly the FormBuilder in mind... :-D Because it's almost impossible to build a Form for me without digging through documentation and/or sourcecode.